src/main/java/bio/terra/service/duos/DuosClient.java

```diff
@@ -46,28 +75,66 @@ String getDatasetUrl(String duosId) {
-   * @throws DuosDatasetNotFoundException if no DUOS dataset found
    */
   public DuosDataset getDataset(String duosId, AuthenticatedUserRequest userRequest) {
     HttpHeaders authedHeaders = getHttpHeaders();
     authedHeaders.setBearerAuth(userRequest.getToken());
+    String url = getDatasetUrl(duosId);
+    logger.info("About to GET {} as authenticated user", url);
     try {
       ResponseEntity<DuosDataset> datasetCall =
           restTemplate.exchange(
-              getDatasetUrl(duosId),
-              HttpMethod.GET,
-              new HttpEntity<>(authedHeaders),
-              DuosDataset.class);
+              url, HttpMethod.GET, new HttpEntity<>(authedHeaders), DuosDataset.class);
       return datasetCall.getBody();
-    } catch (HttpClientErrorException ex) {
-      switch (ex.getStatusCode()) {
-        case BAD_REQUEST -> throw new DuosDatasetBadRequestException(
-            "DUOS dataset identifier %s is malformed".formatted(duosId), ex);
-        case NOT_FOUND -> throw new DuosDatasetNotFoundException(
-            "Could not find DUOS dataset for identifier %s".formatted(duosId), ex);
-        default -> throw ex;
-      }
+    } catch (HttpStatusCodeException ex) {
+      throw convertDuosExToDataRepoEx(ex, duosId);
+    }
+  }
+
+  @VisibleForTesting
+  String getApprovedUsersUrl(String duosId) {
+    return getDatasetUrl(duosId) + "/approved/users";
+  }
+
+  /**
+   * This call to DUOS is performed as the TDR SA rather than a supplied authenticated user.
+   *
+   * @param duosId DUOS dataset ID
+   * @return all users who have access to the dataset via an accepted DAR
+   */
+  public DuosDatasetApprovedUsers getApprovedUsers(String duosId) {
+    HttpHeaders saHeaders = getHttpHeaders();
+    saHeaders.setBearerAuth(googleCredentialsService.getApplicationDefaultAccessToken(SCOPES));
+    String url = getApprovedUsersUrl(duosId);
+    logger.info("About to GET {} as TDR SA", url);
+    try {
+      ResponseEntity<DuosDatasetApprovedUsers> approvedUsersCall =
+          restTemplate.exchange(
+              url, HttpMethod.GET, new HttpEntity<>(saHeaders), DuosDatasetApprovedUsers.class);
+      return approvedUsersCall.getBody();
+    } catch (HttpStatusCodeException ex) {
+      throw convertDuosExToDataRepoEx(ex, duosId);
+    }
+  }
+
+  @VisibleForTesting
+  static ErrorReportException convertDuosExToDataRepoEx(
+      HttpStatusCodeException duosEx, String duosId) {
+    switch (duosEx.getStatusCode()) {
+      case BAD_REQUEST:
+        {
+          return new DuosDatasetBadRequestException(
+              "DUOS dataset identifier %s is malformed".formatted(duosId), duosEx);
+        }
+      case NOT_FOUND:
+        {
+          return new DuosDatasetNotFoundException(
+              "Could not find DUOS dataset for identifier %s".formatted(duosId), duosEx);
+        }
+      default:
+        {
+          return new DuosInternalServerErrorException("Unexpected error from DUOS", duosEx);
+        }
```

Comment:
I think you can replace this with a switch expression
```suggestion
    return switch (duosEx.getStatusCode()) {
      case BAD_REQUEST -> new DuosDatasetBadRequestException(
          "DUOS dataset identifier %s is malformed".formatted(duosId), duosEx);
      case NOT_FOUND -> new DuosDatasetNotFoundException(
          "Could not find DUOS dataset for identifier %s".formatted(duosId), duosEx);
      default -> new DuosInternalServerErrorException("Unexpected error from DUOS", duosEx);
    };
```

Reply:
Ah, thanks! I was struggling with this for some reason, will check it out.

Reply:
I think IntelliJ can do this for you, if the right settings are enabled.